app/main/views/users.py

```diff
@@ -213,6 +214,77 @@ def update_user(user_id):
         abort(400, "Could not update user with: {0}".format(user_update))
 
 
+@main.route('/users/export/<framework_slug>', methods=['GET'])
+def export_users_for_framework(framework_slug):
+
+    # 400 if framework slug is invalid
+    framework = Framework.query.filter(Framework.slug == framework_slug).first()
+    if not framework:
+        abort(400, 'invalid framework')
+
+    if framework.status == 'coming':
+        abort(400, 'framework not yet open')
+
+    supplier_frameworks_and_suppliers_and_users = db.session.query(
+        SupplierFramework, Supplier, User
+    ).join(
+        Framework
+    ).filter(
+        Framework.slug == framework_slug
+    ).filter(
+        SupplierFramework.framework_id == Framework.id
+    ).filter(
+        SupplierFramework.supplier_id == Supplier.supplier_id
+    ).filter(
```

Comment:
For posterity, what we found was that using a `.join()` function gives you an inner join while just listing model names in the `query()` argument returns the cartesian product.